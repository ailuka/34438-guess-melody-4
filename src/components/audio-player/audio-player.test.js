import React from "react";
import renderer from "react-test-renderer";
import AudioPlayer from "./audio-player.jsx";

const mock = {
  song: {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  },
};

describe(`src/audio-player.jsx`, () => {
  describe(`when the song is paused`, () => {
    it(`should render with play button`, () => {
      const {song} = mock;

      const tree = renderer.create(
          <AudioPlayer
            isPlaying={false}
            src={song.src}
            onPlayButtonClick={() => null}
          />, {
            createNodeMock: () => {
              return {};
            },
          }
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe(`when the song is playing`, () => {
    it(`should render with pause button`, () => {
      const {song} = mock;

      const tree = renderer.create(
          <AudioPlayer
            isPlaying={true}
            src={song.src}
            onPlayButtonClick={() => null}
          />, {
            createNodeMock: () => {
              return {};
            },
          }
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
