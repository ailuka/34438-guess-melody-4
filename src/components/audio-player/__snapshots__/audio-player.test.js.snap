// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/audio-player.jsx when the song is paused should render with play button 1`] = `
Array [
  <button
    className="track__button track__button--play"
    disabled={true}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio />
  </div>,
]
`;

exports[`src/audio-player.jsx when the song is playing should render with pause button 1`] = `
Array [
  <button
    className="track__button track__button--pause"
    disabled={true}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio />
  </div>,
]
`;
